File: algos/algos.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

public class Program
{
	public static void Main()
	{
		//Console.WriteLine(fib(6));
		Console.WriteLine(maxChar("wwwwwwwwwwwwwwweeeeeert"));
		//fizzBuzz(21);
		//Console.WriteLine(anagram("OnEeT", "n        Et o e"));
		
	}
	
    static int ASCII_SIZE = 256; 
      
    public static char maxChar(String str) 
    { 
        // Create array to keep the count of 
        // individual characters and  
        // initialize the array as 0 
        int []count = new int[ASCII_SIZE]; 
      
        // Construct character count array 
        // from the input string. 
        int len = str.Length; 
        for (int i = 0; i < len; i++) 
            count[str[i]]++; 
      
        int max = -1; // Initialize max count 
        char result = ' '; // Initialize result 
      
        // Traversing through the string and  
        // maintaining the count of each character 
        for (int i = 0; i < len; i++) { 
            if (max < count[str[i]]) { 
                max = count[str[i]]; 
                result = str[i]; 
            } 
        } 
        return result; 
	}	
		
	public static int reverseInt(int num)
	{
		var absNum= num < 0? num * (-1) : num;
		
		var arr = absNum.ToString().ToCharArray().Reverse();
		var str = String.Join("",arr);
		
		return num < 0 ? Int32.Parse(str)*(-1) : Int32.Parse(str);
	}
	public static string reverse(string str)
	{
		var arr = str.ToCharArray().Reverse();
			return String.Join("",arr);
	}
	public static int fib(int num)
	{
		List<int> res = new List<int>{0,1};
		for(int i= 2; i<=num;i++)
		{
			int a = res[i-1];
			int b = res[i-2];
			res.Add(a+b);
		}
		
		return res[num];
		
	}
	 public static bool IsPalindrome(string str) 
    {
	  str = str.ToLower();
		
		var strArray = str.ToCharArray();
		
		Array.Reverse(strArray);
				
		return (new string(strArray)).Equals(str); 
    }
	public static bool anagram(string strA, string strB){
		if (cleanedString(strA) == cleanedString(strB)) {
			return true;	
		}
		return false;
	}
	public static string cleanedString(string str) {
		Console.WriteLine(str);
		str = str.Replace(" ", "").ToLower();
		Console.WriteLine(str);
		char[] arr = new char[str.Length];
		
		for (int i =0; i<str.Length;i++) {
			arr[i] = str[i];
		}
		
		 Array.Sort(arr);
		string newStr ="";
		foreach(char i in arr )
		{
			newStr+=i;
		}
		Console.WriteLine(newStr);
		return newStr;
	}
	
	public static void fizzBuzz(int num)
	{
		for (int i = 0; i<=num; i++) 
		{
			if( i %3==0 && i %5==0)
			{
				Console.WriteLine("buzzFuzz");
				
			}
			else if ( i % 3==0 )
			{
				Console.WriteLine("buzz");
				
			}
			else if (i %5==0 )
			{
				Console.WriteLine("Fuzz");
				
			}
			else
			{
				Console.WriteLine(i);
				
			}
			
				
		}
	}
}

File: algos/algos.vb
Imports System
Imports System.Collections.Generic

Public Class Program
    Public Shared Sub Main()
        Console.WriteLine(anagram("OnEeT", "n        Et o e"))
    End Sub

    Public Shared Function fib(ByVal num As Integer) As Integer
        Dim res As List(Of Integer) = New List(Of Integer) From {
            0,
            1
        }

        For i As Integer = 2 To num
            Dim a As Integer = res(i - 1)
            Dim b As Integer = res(i - 2)
            res.Add(a + b)
        Next

        Return res(num)
    End Function

    Public Shared Function anagram(ByVal strA As String, ByVal strB As String) As Boolean
        If cleanedString(strA) = cleanedString(strB) Then
            Return True
        End If

        Return False
    End Function

    Public Shared Function cleanedString(ByVal str As String) As String
        Console.WriteLine(str)
        str = str.Replace(" ", "").ToLower()
        Console.WriteLine(str)
        Dim arr As Char() = New Char(str.Length - 1) {}

        For i As Integer = 0 To str.Length - 1
            arr(i) = str(i)
        Next

        Array.Sort(arr)
        Dim newStr As String = ""

        For Each i As Char In arr
            newStr += i
        Next

        Console.WriteLine(newStr)
        Return newStr
    End Function

    Public Shared Sub fizzBuzz(ByVal num As Integer)
        For i As Integer = 0 To num

            If i Mod 3 = 0 AndAlso i Mod 5 = 0 Then
                Console.WriteLine("buzzFuzz")
            ElseIf i Mod 3 = 0 Then
                Console.WriteLine("buzz")
            ElseIf i Mod 5 = 0 Then
                Console.WriteLine("Fuzz")
            Else
                Console.WriteLine(i)
            End If
        Next
    End Sub
End Class
